app/rest/rest/src/main/java/io/syndesis/rest/v1/handler/connection/ConnectorIconHandler.java

```diff
@@ -0,0 +1,157 @@
+import javax.ws.rs.core.StreamingOutput;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static javax.ws.rs.core.HttpHeaders.CONTENT_LENGTH;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+
+@Api(tags = {"connector", "connector-icon"})
+public final class ConnectorIconHandler extends BaseHandler {
+
+    private final Connector connector;
+    private IconDataAccessObject iconDao;
+
+    /* default */ ConnectorIconHandler(final DataManager dataMgr, final Connector connector, final IconDataAccessObject iconDao) {
+        super(dataMgr);
+        this.connector = connector;
+        this.iconDao = iconDao;
+    }
+
+    @POST
+    @ApiOperation("Updates the connector icon for the specified connector and returns the updated connector")
+    @ApiResponses(@ApiResponse(code = 200, response = Connector.class, message = "Updated Connector icon"))
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Connector create(MultipartFormDataInput dataInput) {
+        if (dataInput == null || dataInput.getParts() == null || dataInput.getParts().isEmpty()) {
+            throw new IllegalArgumentException("Multipart request is empty");
+        }
+
+        if (dataInput.getParts().size() != 1) {
+            throw new IllegalArgumentException("Wrong number of parts in multipart request");
+        }
+
+        try {
+            InputPart filePart = dataInput.getParts().iterator().next();
+            InputStream result = filePart.getBody(InputStream.class, null);
+
+            if (result == null) {
+                throw new IllegalArgumentException("Can't find a valid 'file' part in the multipart request");
+            }
+
+            Icon.Builder iconBuilder = new Icon.Builder()
+                .mediaType(filePart.getMediaType().getType());
+
+            Icon icon;
+            String connectorIcon = connector.getIcon();
+            if (connectorIcon != null && connectorIcon.startsWith("db:")) {
+                String connectorIconId = connectorIcon.substring(3);
+                iconBuilder.id(connectorIconId);
+                icon = iconBuilder.build();
+                getDataManager().update(icon);
+            } else {
+                icon = getDataManager().create(iconBuilder.build());
+            }
+
+            iconDao.write(icon.getId().get(), result);
+
+            Connector updatedConnector = new Connector.Builder().createFrom(connector).icon("db:" + icon.getId().get()).build();
+            getDataManager().update(updatedConnector);
+            return updatedConnector;
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Error while reading multipart request", e);
+        }
+    }
+
+    @GET
+    public Response get() {
+        String connectorIcon = connector.getIcon();
+        if (connectorIcon == null) {
+            return Response.status(javax.ws.rs.core.Response.Status.NOT_FOUND).build();
```

Comment:
nit: perhaps static import NOT_FOUND so we don't have the whole package name here? Or just use `Response.Status.NOT_FOUND` like below

Reply:
Done. Was a mess with Response import from xml packages...